node/src/components/consensus/protocols/highway/participation.rs

```diff
@@ -0,0 +1,96 @@
+use std::cmp::Reverse;
+
+use itertools::Itertools;
+
+use crate::{
+    components::consensus::{
+        highway_core::{
+            highway::Highway,
+            state::{Fault, State},
+            validators::ValidatorIndex,
+        },
+        traits::Context,
+    },
+    types::Timestamp,
+};
+
+/// A validator's participation status: whether they are faulty or inactive.
+#[derive(Copy, Clone, Debug, Ord, PartialOrd, Eq, PartialEq)]
+enum Status {
+    LastSeenSecondsAgo(u64),
+    Inactive,
+    EquivocatedInOtherEra,
+    Equivocated,
+}
+
+impl Status {
+    /// Returns a `Status` for a validator unless they are honest and online.
```

Comment:
I think the usage of _unless_ here is wrong, right? It should be "as long as".

Reply:
No, this only returns `Some` for validators that are _not_ honest and online. We don't list the honest and online ones here, so it's easier to see if something's wrong (and who's to blame).

Reply:
Synonyms for _unless_ are:

> and, barely, besides, but, except, further, however, just, merely, moreover, nevertheless, notwithstanding, notwithstanding, only, provided, save, still, that, tho, yet.

so this comment means that we will return `Status` if they're not honest and online but it's the other way around, isn't it?

Reply:
No, it's exactly that:
* If they're honest and online, we return `None`.
* If they're _not_ honest and online, we return some `Status`.